team/bundles/org.eclipse.core.pki/src/org/eclipse/pki/auth/AuthenticationPlugin.java

```diff
@@ -0,0 +1,1133 @@
+package org.eclipse.pki.auth;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.KeyStore.Builder;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.List;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.KeyStoreBuilderParameters;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.PasswordCallback;
+
+import org.eclipse.core.pki.FingerprintX509;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.ILog;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.preferences.ConfigurationScope;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+import org.eclipse.jface.preference.IPersistentPreferenceStore;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.osgi.framework.BundleContext;
+
+import org.eclipse.pki.dialog.PassphraseDialog;
+import org.eclipse.pki.exception.UserCanceledException;
+import org.eclipse.pki.jsse.CdeX509TrustManager;
+import org.eclipse.pki.pkcs.VendorImplementation;
+import org.eclipse.pki.pkiselection.PKCSSelected;
+import org.eclipse.pki.pkiselection.PKCSpick;
+import org.eclipse.pki.pkiselection.PKIProperties;
+import org.eclipse.pki.preferences.AuthenticationPreferences;
+import org.eclipse.pki.util.KeyStoreFormat;
+import org.eclipse.pki.util.KeyStoreUtil;
+import org.eclipse.pki.util.LogUtil;
+import org.eclipse.pki.util.PKIAuthenticator;
+import org.eclipse.pki.util.SecurityFileSnapshot;
+import org.eclipse.pki.wizard.PKILoginWizard;
+import org.eclipse.pki.wizard.TrustStoreLoginWizard;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.preferences.ConfigurationScope;
+
+/**
+ * The activator class controls the plug-in life cycle
+ */
+public class AuthenticationPlugin extends AbstractUIPlugin {

```

Comment:
this one should be split to `core` and `ui` part

Reply:
Agree ton of work needed here.